### PR TITLE
beauty: full conan v2 support + add test_v1_package

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
ohoh, very bad: https://github.com/dfleury2/beauty/blob/1.0.0-rc1/CMakeLists.txt#L6-L20

File: recipes/beauty/all/conanfile.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
-import os
-from conans import ConanFile, tools
-from conan.tools.cmake import CMake
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, rmdir
```
Comment (SpaceIm) [inline recipes/beauty/all/conanfile.py:5]:
```suggestion
from conan.tools.files import apply_conandata_patches, copy, get, rmdir
```

File: recipes/beauty/all/conanfile.py [hunk 2]
```diff
@@ -15,24 +17,26 @@ class BeautyConan(ConanFile):
...
+    generators = "CMakeDeps"
 
     @property
     def _compilers_minimum_version(self):
         return {
             "gcc": "8",
             "clang": "11",
             "Visual Studio": "16",
+            "msvc": "192",
         }
```
Comment (SpaceIm) [inline recipes/beauty/all/conanfile.py:41]:
```suggestion

    def export_sources(self):
        for p in self.conan_data.get("patches", {}).get(self.version, []):
            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)

```

File: recipes/beauty/all/conanfile.py [hunk 3]
```diff
@@ -58,25 +66,34 @@ def validate(self):
...
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.generate()
 
     def build(self):
         cmake = CMake(self)
```
Comment (SpaceIm) [inline recipes/beauty/all/conanfile.py:84]:
```suggestion
        apply_conandata_patches(self)
        cmake = CMake(self)
```

**Review by uilianries — APPROVED** [review]

**Review by toge — APPROVED** [review]

**Pull request merged**